ios/ReactNativeHealthkit.swift

```diff
@@ -657,17 +683,71 @@ class ReactNativeHealthkit: RCTEventEmitter {
                             "sourceRevision": serializeSourceRevision(_sourceRevision: workout.sourceRevision) as Any
                         ]
 
+                        //this is used for our laps functionality to get markers 
+                        //https://developer.apple.com/documentation/healthkit/hkworkoutevent
+                        if let events = workout.workoutEvents {
+                            var eventDicts: [[String: Any]] = []
+                            for event in events {
+                                let eventStartDate = self._dateFormatter.string(from: event.dateInterval.start)
+                                let eventEndDate = self._dateFormatter.string(from: event.dateInterval.end)
+                                let eventDict: [String: Any] = [
+                                    "type": event.type.rawValue, //https://developer.apple.com/documentation/healthkit/hkworkouteventtype
+                                    "startDate": eventStartDate,
+                                    "endDate": eventEndDate
+                                ]
+                                eventDicts.append(eventDict)
+                            }
+                            dict["events"] = eventDicts
+                        }
+                        
+                        //also used for our laps functionality to get activities for custom workouts defined by the user
+                        //https://developer.apple.com/documentation/healthkit/hkworkout/1615340-init 
+                        //it seems this might be depricated in the latest beta so this might need updating!
+                        var activitiesDicts: [[String: Any]] = []
+                        if #available(iOS 16.0, *) {
+                            let activities: [HKWorkoutActivity] = workout.workoutActivities
+                            
+                            if !activities.isEmpty{
+                                for activity in activities {
+                                    var activityStartDate = ""
+                                    var activityEndDate = ""
+                                    if let start = activity.startDate as Date? {
+                                        activityStartDate = self._dateFormatter.string(from: activity.startDate)
+                                    }
+                                    let activityDict: [String: Any] = [
+                                        "startDate": activityStartDate,
```

Comment:
Feels like uuid, endDate and duration should also be exposed here?

Reply:
done